sdks/java/core/src/test/java/com/google/cloud/dataflow/sdk/transforms/ParDoTest.java

```diff
@@ -1523,20 +1528,66 @@ public void testMutatingInputCoderDoFnError() throws Exception {
   }
 
   @Test
-  public void testIncludesDoFnDisplayData() {
-    Bound<String, String> parDo =
-        ParDo.of(
-            new DoFn<String, String>() {
-              @Override
-              public void processElement(ProcessContext c) {}
+  public void testDoFnDisplayData() {
+    DoFn<String, String> fn = new DoFn<String, String>() {
+      @Override
+      public void processElement(ProcessContext c) {
+      }
 
-              @Override
-              public void populateDisplayData(Builder builder) {
-                builder.add("foo", "bar");
-              }
-            });
+      @Override
+      public void populateDisplayData(Builder builder) {
+        builder.add("doFnMetadata", "bar");
+      }
+    };
+
+    Bound<String, String> parDo = ParDo.of(fn);
+
+    DisplayData displayData = DisplayData.from(parDo);
+    assertThat(displayData, hasDisplayItem(allOf(
+        hasKey("fn"),
+        hasType(DisplayData.Type.JAVA_CLASS),
+        hasValue(fn.getClass().getName()))));
+
+    assertThat(displayData, includes(fn));
+  }
+
+  @Test
+  public void testDoFnWithContextDisplayData() {
+    DoFnWithContext<String, String> fn = new DoFnWithContext<String, String>() {
+      @ProcessElement
+      public void proccessElement(ProcessContext c) {}
+
+      @Override
+      public void populateDisplayData(Builder builder) {
+        builder.add("fnMetadata", "foobar");
+      }
+    };
+
+    Bound<String, String> parDo = ParDo.of(fn);
+
+    DisplayData displayData = DisplayData.from(parDo);
+    assertThat(displayData, includes(fn));
```

Comment:
I see where you're coming from. I'd prefer to make the Matcher method self-describing rather than require a comment on each usage. Maybe something like:

    assertThat(displayData, includesDisplayDataFrom(fn));

Let me take this as a follow-up item once the pending PRs go through in order to avoid merge conflicts. Tracking in [BEAM-195](https://issues.apache.org/jira/browse/BEAM-195)